qiskit_experiments/framework/experiment_data.py

```diff
@@ -1534,6 +1579,51 @@ def cancel(self) -> bool:
         jobs_cancelled = self.cancel_jobs()
         return analysis_cancelled and jobs_cancelled
 
+    def save_done(self) -> bool:
+        """Checks whether all database save jobs were completed successfully
+
+        Returns:
+            Whether all the analysis results/figure upload jobs are done,
+            as opposed to running/cancelled/failed.
+        """
+        with self._save_futures.lock:
+            for fut in self._save_futures.values():
+                if not fut.done():
+                    return False
+        return True
+
+    def block_for_save(self, timeout: Optional[float] = None) -> "ExperimentData":
+        """Block until all pending save operations finish.
+
+        Args:
+            timeout: Timeout in seconds for waiting for results.
+
+        Returns:
+            The experiment data with finished jobs and post-processing.
+        """
+        start_time = time.time()
+        with self._save_futures.lock:
+            save_ids = self._save_futures.keys()
+            save_futs = self._save_futures.values()
+        self._wait_for_futures(save_futs, name="database save", timeout=timeout)
+        # Clean up done job futures
+        num_saves = len(save_futs)
+        for jid, fut in zip(save_ids, save_futs):
+            if (fut.done() and not fut.exception()) or fut.cancelled():
+                del self._save_futures[jid]
+                num_saves -= 1
+
+        # Check if more futures got added while this function was running
+        # and block recursively. This could happen if an analysis callback
+        # spawns another callback or creates more jobs
+        if len(self._save_futures) > num_saves:
+            time_taken = time.time() - start_time
+            if timeout is not None:
+                timeout = max(0, timeout - time_taken)
+            return self.block_for_results(timeout=timeout)
```

Comment:
Do we need to `block_for_results` or `block_for_save` here?

Reply:
Yes, that's a silly bug. Fixing it.